### PR TITLE
[PW 8347] Support 6.5

### PR DESCRIPTION
**Pull request by candemiralp**
## Summary

Added support for affected changes in [Shopware 6.5](https://github.com/shopware/platform/blob/trunk/UPGRADE-6.5.md)
Support is backwards compatible with all 6.4 versions

## Tested scenarios

- Card payments (incl 3DS, 3DS2)
- Redirect payments
- Wallet payments
- Donations
- Giftcard full payments
- Giftcard partial payments
- Oneclick stored payments

### CODE REVIEW

File: .github/workflows/e2e-test.yml [hunk 1]
```diff
@@ -22,15 +22,15 @@ jobs:
 
       - name: Initiate Dockware
         run: |
-          docker pull dockware/dev:6.4.20.1
+          docker pull dockware/dev:latest
```
Comment (hossam-adyen) [inline .github/workflows/e2e-test.yml:25]: I think it's better for the longrun to bind to a specific version
```suggestion
          docker pull dockware/dev:6.5
```

File: .github/workflows/e2e-test.yml [hunk 2]
```diff
@@ -22,15 +22,15 @@ jobs:
...
           docker run --rm -p 443:443 --name shopware6 \
             --mount type=bind,source="$(pwd)",target=/data/extensions/workdir \
-            --env PHP_VERSION=7.4 -d dockware/dev:6.4.20.1
+            --env PHP_VERSION=8.2 -d dockware/dev:latest
           sleep 30
           docker logs shopware6
-          docker exec shopware6 bash -c "mysql -u root -proot shopware -e \"UPDATE sales_channel_domain SET url='https://local.shopware.shop' WHERE url NOT LIKE 'default.%';\""
+          docker exec shopware6 bash -c "sudo mysql -u root -proot shopware -e \"UPDATE sales_channel_domain SET url='https://local.shopware.shop' WHERE url NOT LIKE 'default.%';\""
```
Comment (hossam-adyen) [inline .github/workflows/e2e-test.yml:31]: why now we need sudo and before we didn't?

File: src/Storefront/Controller/FrontendProxyController.php [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+    public function __construct(
+        AbstractCartOrderRoute $cartOrderRoute,
+        AbstractHandlePaymentMethodRoute $handlePaymentMethodRoute,
+        AbstractContextSwitchRoute $contextSwitchRoute,
+        CartService $cartService,
+        PaymentController $paymentController,
+        OrderApiController $orderApiController,
+        DonateController $donateController
+    )
+    {
```
Comment (candemiralp) [inline src/Storefront/Controller/FrontendProxyController.php:71]:
```suggestion
    ) {
```

**Review by RokPopov — APPROVED** [review]

File: .github/workflows/e2e-test.yml (hunk 2 above)
Comment (peterojo) [reply .github/workflows/e2e-test.yml:31]: New docker image for 6.5 doesn't allow GH action user to connect to mysql

**Pull request merged**